lib/cocoapods/target/build_settings.rb

```diff
@@ -607,7 +607,7 @@ def initialize(target, test_spec = nil)
 
           libraries = []
           if test_xcconfig? || target.requires_frameworks? && !target.static_framework?
-            libraries.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') }
+            libraries.concat vendored_static_libraries.map { |l| File.basename(l, l.extname).sub(/\Alib/, '') }
```

Comment:
the logic of `MyLibrary` -> `libMyLibrary.a` also existing in `:static_libraries_to_import` and `:dynamic_libraries_to_import` - could you pull it out into a function?

Reply:
Sure, done